core/shared/src/main/scala/zio/ZPool.scala

```diff
@@ -47,74 +47,90 @@ object ZPool {
+  def fromIterable[A](iterable: => Iterable[A])(implicit trace: ZTraceElement): ZIO[Scope, Nothing, ZPool[Nothing, A]] =
     for {
-      iterable <- ZManaged.succeed(iterable)
-      source   <- Ref.make(iterable.toList).toManaged
+      iterable <- ZIO.succeed(iterable)
+      source   <- Ref.make(iterable.toList)
       get = if (iterable.isEmpty) ZIO.never
             else
               source.modify {
                 case head :: tail => (head, tail)
                 case Nil          => throw new IllegalArgumentException("No items in list!")
               }
-      pool <- ZPool.make(ZManaged.fromZIO(get), iterable.size)
+      pool <- ZPool.make(get, iterable.size)
     } yield pool
 
   /**
    * Makes a new pool of the specified fixed size. The pool is returned in a
-   * `Managed`, which governs the lifetime of the pool. When the pool is
-   * shutdown because the `Managed` is used, the individual items allocated by
-   * the pool will be released in some unspecified order.
+   * `Scope`, which governs the lifetime of the pool. When the pool is shutdown
+   * because the `Scope` is closed, the individual items allocated by the pool
+   * will be released in some unspecified order.
    */
-  def make[R, E, A](get: => ZManaged[R, E, A], size: => Int)(implicit trace: ZTraceElement): URManaged[R, ZPool[E, A]] =
+  def make[R, E, A](get: => ZIO[R, E, A], size: => Int)(implicit
+    trace: ZTraceElement
+  ): ZIO[R with Scope, Nothing, ZPool[E, A]] =
     for {
-      size <- ZManaged.succeed(size)
+      size <- ZIO.succeed(size)
       pool <- makeWith(get, size to size)(Strategy.None)
     } yield pool
 
   /**
    * Makes a new pool with the specified minimum and maximum sizes and time to
    * live before a pool whose excess items are not being used will be shrunk
-   * down to the minimum size. The pool is returned in a `Managed`, which
-   * governs the lifetime of the pool. When the pool is shutdown because the
-   * `Managed` is used, the individual items allocated by the pool will be
-   * released in some unspecified order.
+   * down to the minimum size. The pool is returned in a `Scope`, which governs
+   * the lifetime of the pool. When the pool is shutdown because the `Scope` is
+   * used, the individual items allocated by the pool will be released in some
+   * unspecified order.
    * {{{
-   * ZPool.make(acquireDbConnection, 10 to 20, 60.seconds).use { pool =>
-   *   pool.get.use {
-   *     connection => useConnection(connection)
+   * ZIO.scoped {
+   *   ZPool.make(acquireDbConnection, 10 to 20, 60.seconds).flatMap { pool =>
+   *     ZIO.scoped {
+   *       pool.get.flatMap {
+   *         connection => useConnection(connection)
+   *       }
+   *     }
    *   }
    * }
    * }}}
    */
-  def make[R, E, A](get: => ZManaged[R, E, A], range: => Range, timeToLive: => Duration)(implicit
+  def make[R, E, A](get: => ZIO[R, E, A], range: => Range, timeToLive: => Duration)(implicit
```

Comment:
@adamgfraser
Could you please check if the type of the `get` parameter is properly migrated?
I think it should be `ZIO[R with Scope, E, A]`

Reply:
I agree that would be the most straightforward translation. Here I think we can do this for a couple reasons:

1. We aren't looking to eliminate the scope. When we want to eliminate the scope we definitely need to use something like `R with Scope` so we have a type `R` to refer to that has all of the requirements of the workflow except the scope. In this case our return type has a `Scope` no matter what.
2. We are already providing `get` with our own `Scope` in `get.provideSomeEnvironment[Scope](env.union[Scope](_))`. The right hand side wins in `union` so if `get` does require a `Scope` we are providing it with our scope, which is what we want. And if it doesn't we are just providing it with a superfluous scope which is harmless.
3. It significantly improves type inference.

This is definitely a tricky case so if you see anything that looks off we can take a look at it but I just ran a few tests myself and everything seemed to be in proper order. Items are returned to the pool when scope of `Pool#get` is closed and the resources themselves are released when the scope of the `Pool` is closed.